hs_core/tasks.py

```diff
@@ -166,6 +167,15 @@ def create_bag_by_irods(resource_id, istorage=None):
     from hs_core.hydroshare.utils import get_resource_by_shortkey
 
     res = get_resource_by_shortkey(resource_id)
+    try:
+        # we will create xml metadata files every time prior to creating the bag
+        # the assumption here is that bag is being created due to the resource
+        # modified (AVU)
+        create_bag_files(res, fed_zone_home_path=res.resource_federation_path)
+    except Exception as ex:
+        logger.error('Failed to create bag files. Error:{}'.format(ex.message))
+        return False
+

```

Comment:
I think it is better to generate xml files whenever we are creating the bag to guarantee that the xml file content is correct/up-to-date. I can also add the additional check if it is absolutely needed.